nexus/db-queries/src/db/datastore/deployment.rs

```diff
@@ -403,6 +495,56 @@ impl DataStore {
             zones_config.sort();
         }
 
+        // Load all the physical disks for each sled.
+        {
+            use db::schema::bp_omicron_physical_disk::dsl;
+
+            let mut paginator = Paginator::new(SQL_BATCH_SIZE);
+            while let Some(p) = paginator.next() {
+                // `paginated` implicitly orders by our `id`, which is also
+                // handy for testing: the physical disks are always consistently ordered
+                let batch = paginated(
+                    dsl::bp_omicron_physical_disk,
+                    dsl::id,
+                    &p.current_pagparams(),
+                )
+                .filter(dsl::blueprint_id.eq(blueprint_id))
+                .select(BpOmicronPhysicalDisk::as_select())
+                .load_async(&*conn)
+                .await
+                .map_err(|e| {
+                    public_error_from_diesel(e, ErrorHandler::Server)
+                })?;
+
+                paginator = p.found_batch(&batch, &|d| d.id);
+
+                for d in batch {
+                    let sled_disks = blueprint_disks
+                        .get_mut(&TypedUuid::<SledKind>::from_untyped_uuid(
+                            d.sled_id,
+                        ))
+                        .ok_or_else(|| {
+                            // This error means that we found a row in
+                            // bp_omicron_physical_disk with no associated record in
+                            // bp_sled_omicron_physical_disks.  This should be
+                            // impossible and reflects either a bug or database
+                            // corruption.
+                            Error::internal_error(&format!(
+                                "disk {:?}: unknown sled: {:?}",
```

Comment:
Small nit:

```suggestion
                                "disk {}: unknown sled: {}",
```

For `Uuid` this doesn't matter (its display and debug impls are [identical](https://docs.rs/uuid/1.8.0/src/uuid/fmt.rs.html#22-33)), but if/when we move these to `TypedUuid` it does matter (TypedUuid [includes the tag](https://docs.rs/newtype-uuid/1.0.1/src/newtype_uuid/lib.rs.html#155-158) in its debug impl, which I think we generally don't want in logs / error messages).